src/hotkeys/hotkeys-service.ts

```diff
@@ -1,245 +1,168 @@
+    eventType: HotKeyEventTypes = "keydown",
+    options: Partial<Omit<IHotKeyListener, "listener">> = {}
   ): HotkeysService {
-    const key: string = this.prepareKey(hotkeys);
-    if (!this.store.get(eventType)) {
-      this.store.set(eventType, new Map());
-    }
-    const store = this.store.get(eventType) as HotkeysStoreType;
-    if (!store.get(key)) {
-      store.set(key, new HandlerList());
-    }
-    const list: HandlerList = store.get(key) as HandlerList;
+    const storeKey = this.convertKeysToStoreKey(hotkeys);
+
+    const handlerStore: HotkeysStoreType =
+      this.handlerStore.get(eventType) || new Map();
+    const list = handlerStore.get(storeKey) || new HotKeyListenerList();
+
     list.add({
       listener,
       ignoreNamespace: options.ignoreNamespace || false,
       ignoreFocusedElements: options.ignoreFocusedElements || false,
       namespace: options.namespace || "",
       description: options.description || ""
     });
-    return this;
-  }
 
-  /**
-   * Replace hotkey listener to the store
-   * @param {string} hotkeys
-   * @param {Listener} oldListener
-   * @param {Listener} listener
-   * @param {EventType} eventType keydown or keyup
-   * @param {string|undefined} namespace
-   * @returns {HotkeysService}
-   */
-  replaceListener(
-    hotkeys: string,
-    oldListener: Listener,
-    listener: Listener,
-    eventType: EventType = "keydown",
-    namespace = ""
-  ): HotkeysService {
-    const key: string = this.prepareKey(hotkeys);
-    const store = this.store.get(eventType) as HotkeysStoreType;
-    if (!store) {
-      return this;
-    }
-    if (!store.get(key)) {
-      return this;
-    }
-    const list: HandlerList = store.get(key) as HandlerList;
-    list.replace({ listener: oldListener, namespace }, { listener, namespace });
+    this.handlerStore.set(eventType, handlerStore.set(storeKey, list));
     return this;
   }
 
-  /**
-   * Removes listener from store
-   * @param {string} hotkeys
-   * @param {Listener} listener
-   * @param {EventType} eventType keydown or keyup
-   * @param {string} namespace
-   * @returns {HotkeysService}
-   */
   remove(
     hotkeys: string,
-    listener: Listener,
-    eventType: EventType = "keydown",
+    listener: HotKeyListener,
+    eventType: HotKeyEventTypes = "keydown",
     namespace = ""
   ): HotkeysService {
-    const key: string = this.prepareKey(hotkeys);
-    const store = this.store.get(eventType);
-    if (!store) {
+    const storeKey = this.convertKeysToStoreKey(hotkeys);
+
+    const handlerStore = this.handlerStore.get(eventType);
+    if (!handlerStore) {
       return this;
     }
-    const list: HandlerList | undefined = store.get(key);
+
+    const list = handlerStore.get(storeKey);
     if (list) {
       list.remove({ listener, namespace });
     }
+
     return this;
   }
 
-  /**
-   * Returns list of hotkeys with description
-   * @returns {IHotkeysPreview}
-   */
   getHotkeysWithDescriptions(): IHotkeysPreview {
```

Comment:
@abarabash89  this method should be moved to the hotkeys popup. It should be placed here.

Reply:
👍